### PR TITLE
Add Acceptance test verification for HIP-873 decimals field

### LINKED ISSUE #7838: Add acceptance tests for HIP-873 decimals field
### Problem

The work for [HIP-873](https://github.com/hashgraph/hedera-mirror-node/pull/7699) added the `decimals` and `name` field to 
`/api/v1/tokens`

 And the `decimals` field to 
`/api/v1/tokens/{tokenId}/balances`
`/api/v1/accounts/{idOrAliasOrEvmAddress}/tokens`

### Solution

Add verification of these fields to the existing acceptance tests

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by edwin-greene**
**Description**:
Adds verification to Acceptance test for the HIP-873 `decimals` and `name` fields.

**Related issue(s)**:

Fixes #7838 

**Notes for reviewer**:
Ran `token.feature`, `tokenAllowance.feature` and `call.feature` against previewnet to verify acceptance tests pass.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java [hunk 1]
```diff
@@ -100,6 +101,11 @@ public void createNamedToken(TokenNameEnum tokenName) {
             verifyMirrorTransactionsResponse(mirrorClient, 200);
         }
         var tokenInfo = mirrorClient.getTokenInfo(tokenAndResponse.tokenId().toString());
+        assertNotNull(tokenInfo.getDecimals());
```
Comment (steven-sheehy) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java:104]: We should only use assertThat for new changes

Comment (edwin-greene) [reply hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java]: Thanks, I have updated the asserts.

**Review by steven-sheehy — DISMISSED** [review]
LGTM

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java [hunk 2]
```diff
@@ -100,6 +101,11 @@ public void createNamedToken(TokenNameEnum tokenName) {
             verifyMirrorTransactionsResponse(mirrorClient, 200);
         }
         var tokenInfo = mirrorClient.getTokenInfo(tokenAndResponse.tokenId().toString());
+        assertThat(tokenInfo.getDecimals()).isNotNull();
+        if (tokenName.getTokenType() == TokenType.NON_FUNGIBLE_UNIQUE) {
+            assertThat(tokenInfo.getDecimals()).isEqualTo("0");
+        }
+        assertThat(tokenInfo.getName()).isNotNull();
```
Comment (xin-hedera) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java:108]: this doesn't verify what's outlined in the ticket right?

name / decimals are there long before HIP-873.

we are supposed to verify the two fields exist in the response of `/api/v1/tokens`

Comment (edwin-greene) [reply hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/TokenFeature.java]: Thanks Xin, I have updated it to test the correct decimals.

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/MirrorNodeClient.java [hunk 3]
```diff
@@ -269,11 +268,21 @@ public Schedule getScheduleInfo(String scheduleId) {
...
+        log.debug("Verify token balances '{}' is returned by Mirror Node", tokenId);
+        return callRestEndpoint("/tokens/{tokenId}/balances", TokenBalancesResponse.class, tokenId);
+    }
+
     public TokenInfo getTokenInfo(String tokenId) {
         log.debug("Verify token '{}' is returned by Mirror Node", tokenId);
         return callRestEndpoint("/tokens/{tokenId}", TokenInfo.class, tokenId);
     }
 
+    public TokensResponse getTokensResponse(String tokenId) {
```
Comment (xin-hedera) [inline hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/MirrorNodeClient.java:281]: nit, keep naming consistent

```suggestion
    public TokensResponse getTokens(String tokenId) {
```

**Review by xin-hedera — APPROVED** [review]
LGTM

File: hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/MirrorNodeClient.java (hunk 3 above)
Comment (edwin-greene) [reply hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/client/MirrorNodeClient.java]: Thanks I have updated it

**Review by steven-sheehy — APPROVED** [review]

**Review by xin-hedera — APPROVED** [review]
LGTM

**Pull request merged**